src/server/pkg/storage/fileset/storage.go

```diff
@@ -126,6 +126,12 @@ func (s *Storage) Open(ctx context.Context, fileSets []string, opts ...index.Opt
 	return newMergeReader(s.chunks, fss), nil
 }
 
+// TODO: public?
+func (s *Storage) exists(ctx context.Context, fileset string) bool {
```

Comment:
This type of check seems sketchy. Interpreting any error as doesn't exist seems error prone.

Reply:
Yeah I was thinking about that. We do this in the obj.Client.  I'll change it to return `(bool, error)`

Reply:
We should probably do the same for our object clients at some point.